engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/RollupNodeOperationsRecorder.java

```diff
@@ -43,7 +55,35 @@ RollupTable.NodeType getNodeType() {
 
     @Override
     public boolean isEmpty() {
-        return super.isEmpty();
+        return super.isEmpty() && recordedUpdateViews.isEmpty();
+    }
+
+    Collection<? extends SelectColumn> getRecordedUpdateViews() {
+        return recordedUpdateViews;
+    }
+
+    static Table applyUpdateViews(
+            @Nullable final RollupNodeOperationsRecorder nodeOperations,
+            @NotNull final Table input) {
+        if (nodeOperations != null && !nodeOperations.getRecordedUpdateViews().isEmpty()) {
+            return input.updateView(nodeOperations.getRecordedUpdateViews());
+        }
+        return input;
+    }
+
+    @Override
+    TableDefinition getResultDefinition() {
+        // Get the base results and apply recorded updateViews to get the final result definition
+        final TableDefinition baseDefinition = super.getResultDefinition();
```

Comment:
Are we doing format vs. update views in the right order now?

Reply:
Yes, this change doesn't affect evaluation, just the generation of the result definition.